File: src/dataset/go/grains_Exercism_go_rootulp/openAI/ChatGPT4_grains_v2.go
package grains

import "errors"

// Square returns the number of grains on the provided square.
func Square(n int) (uint64, error) {
	if n < 1 || n > 64 {
		return 0, errors.New("square: n is not between 1 and 64 (inclusive)")
	}
	return 1 << (n - 1), nil
}

// Total returns the sum of all grains on the chess board.
func Total() uint64 {
	// The total is a constant value: 2^64 - 1
	return (1 << 64) - 1
}

File: src/dataset/c/grains_Exercism_m3g4d1v3r/openAI/ChatGPT4_grains_v2.c
#include "grains.h"

static uint64_t board[BOARD_SIZE + 1] = {0};
static bool board_initialized = false;

uint64_t square(uint8_t index) {
    if (index == 0 || index > BOARD_SIZE) return 0;
    return (uint64_t)1 << (index - 1);
}

uint64_t total(void) {
    if (!board_initialized) {
        board[1] = 1;
        for (uint64_t i = 2; i <= BOARD_SIZE; i++) {
            board[i] = board[i - 1] * 2;
        }
        board_initialized = true;
    }
    return ((uint64_t)1 << BOARD_SIZE) - 1;
}